core/src/main/scala/kafka/server/BrokerToControllerChannelManagerImpl.scala

```diff
@@ -125,15 +129,25 @@ class BrokerToControllerChannelManagerImpl(metadataCache: kafka.server.MetadataC
   }
 
   override def sendRequest(request: AbstractRequest.Builder[_ <: AbstractRequest],
-                           callback: RequestCompletionHandler): Unit = {
-    requestQueue.put(BrokerToControllerQueueItem(request, callback))
+                           callback: ControllerRequestCompletionHandler,
+                           retryDeadlineMs: Long): Unit = {
+    requestQueue.put(BrokerToControllerQueueItem(request, callback, retryDeadlineMs))
     requestThread.wakeup()
   }
+}
+
+abstract class ControllerRequestCompletionHandler extends RequestCompletionHandler {
 
+  /**
+   * Fire when the request transmission time passes the caller defined deadline on the channel queue.
+   * This is different from the original request's timeout.
```

Comment:
Can we explain how it is different? We want to emphasize that it covers the total time including retries which might be the result of request timeouts.